astropy/timeseries/periodograms/lombscargle/implementations/scipy_impl.py

```diff
@@ -57,7 +57,8 @@ def lombscargle_scipy(t, y, frequency, normalization="standard", center_data=Tru
         y = y - y.mean()
 
     # Note: scipy input accepts angular frequencies
-    p = signal.lombscargle(t, y, 2 * np.pi * frequency)
+    # floating_mean=True will fail tests due to https://github.com/scipy/scipy/pull/21277
+    p = signal.lombscargle(t, y, 2 * np.pi * frequency, floating_mean=False)
```

Comment:
This will fail with all non-dev versions of scipy anyway, where the option does not exist.

Reply:
Easy to add a scipy ver check here if this is how we want things to happen.

Reply:
Ver check added. Thanks for the review!